assertj-core/src/main/java/org/assertj/core/internal/CronExpression.java

```diff
@@ -0,0 +1,122 @@
+package org.assertj.core.internal;
+
+import org.assertj.core.api.AssertionInfo;
+
+import static org.assertj.core.error.ShouldCronExpressionBeValid.shouldCronExpressionBeValid;
+import static org.assertj.core.internal.CronUnit.*;
+
+/**
+ * CronExpression contains a cronable string that to set up a scheduled task.
+ * @author Neil Wang
+ */
+public class CronExpression {
```

Comment:
Related to my comment, having this new type in the internal package and expose it as part of the public API like in `assertThat(CronExpression)` will cause trouble in module-path and OSGi based applications.